marathon_acme/tests/test_sse_protocol.py

```diff
@@ -1,295 +1,238 @@
         on those characters.
         """
-        data = []
-        self.protocol.set_callback('message', data.append)
         self.protocol.dataReceived(b'data:hello\ndata:world\r\r\n')
 
-        assert_that(data, Equals(['hello\nworld']))
+        assert_that(self.messages, Equals([('message', 'hello\nworld')]))
 
     def test_empty_data(self):
         """
         When the data field is specified in an event but no data is given, the
-        callback should receive an empty string.
+        handler should receive a message with empty data.
         """
-        data = []
-        self.protocol.set_callback('message', data.append)
         self.protocol.dataReceived(b'data:\r\n\r\n')
 
-        assert_that(data, Equals(['']))
+        assert_that(self.messages, Equals([('message', '')]))
 
     def test_no_data(self):
         """
         When the data field is not specified and the event is completed, the
-        callback should not be called.
+        handler should not be called.
         """
-        data = []
-        self.protocol.set_callback('message', data.append)
         self.protocol.dataReceived(b'\r\n')
 
-        assert_that(data, Equals([]))
+        assert_that(self.messages, Equals([]))
 
     def test_space_before_value(self):
         """
         When a field/value pair is received, and there is a space before the
         value, the leading space should be stripped.
         """
-        data = []
-        self.protocol.set_callback('message', data.append)
         self.protocol.dataReceived(b'data: hello\r\n\r\n')
 
-        assert_that(data, Equals(['hello']))
+        assert_that(self.messages, Equals([('message', 'hello')]))
 
     def test_space_before_value_strip_only_first_space(self):
         """
         When a field/value pair is received, and there are multiple spaces at
         the start of the value, the leading space should be stripped and the
         other spaces left intact.
         """
-        data = []
-        self.protocol.set_callback('message', data.append)
         self.protocol.dataReceived(b'data:%s\r\n\r\n' % (b' ' * 4,))
 
-        assert_that(data, Equals([' ' * 3]))
+        assert_that(self.messages, Equals([('message', ' ' * 3)]))
 
     def test_custom_event(self):
         """
-        If a custom event is set for an event, a callback for that event should
-        be called.
+        If a custom event is set for an event, a the handler for should be
+        called with the correct event.
         """
-        data = []
-        self.protocol.set_callback('my_event', data.append)
         self.protocol.dataReceived(b'event:my_event\r\n')
         self.protocol.dataReceived(b'data:hello\r\n\r\n')
 
-        assert_that(data, Equals(['hello']))
-
-    def test_multiple_events(self):
-        """
-        If a multiple different event types are received, the callbacks for
-        each of those events should be called.
-        """
```

Comment:
It seemed less relevant now that the protocol doesn't take multiple callbacks. But maybe I should leave it/update it rather.